dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -556,8 +571,22 @@ public Boolean checkCWL(File content){
                     }
                 }
             }
-            if(versionFound && classFound){
+            if(inputFound && outputFound && classFound){
+                if(!commandFound){
+                    out("Warning: 'baseCommand' field is missing in the entry file.");
+                }
+                if(!versionFound){
+                    out("Warning: 'cwlVersion' field is missing in the entry file.");
+                }
                 return true;
+            } else if(!inputFound && !outputFound && !classFound){
+                return false;
```

Comment:
Why is there an error message when all three are missing? 
Also, this kind of structure means that you get only warning if, say, both outputs and inputs are missing. You should restructure this to get two warnings if two sections are missing.